contracts/truefi2/DebtToken.sol

```diff
@@ -4,37 +4,108 @@ pragma solidity 0.6.10;
 import {ERC20} from "../common/UpgradeableERC20.sol";
 import {IDebtToken} from "./interface/ILoanToken2.sol";
 import {ITrueFiPool2} from "./interface/ITrueFiPool2.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 
 contract DebtToken is IDebtToken, ERC20 {
-    function borrower() external override view returns (address) {}
-
-    function amount() external override view returns (uint256) {}
-
-    function debt() external override view returns (uint256) {}
-
-    function pool() external override view returns (ITrueFiPool2) {}
-
-    function status() external override view returns (Status) {}
-
-    function redeem(uint256 _amount) external override {}
-
-    function repay(address _sender, uint256 _amount) external override {}
-
-    function repayInFull(address _sender) external override {}
-
-    function reclaim() external override {}
-
-    function liquidate() external override {}
-
-    function repaid() external override view returns (uint256) {}
-
-    function isRepaid() external override view returns (bool) {}
-
-    function balance() external override view returns (uint256) {}
-
-    function value(uint256 _balance) external override view returns (uint256) {}
-
-    function token() external override view returns (ERC20) {}
-
-    function version() external override pure returns (uint8) {}
+    using SafeMath for uint256;
+    using SafeERC20 for ERC20;
+
+    address public override borrower;
+    uint256 public override debt;
+
+    address public liquidator;
+    ITrueFiPool2 public override pool;
+    ERC20 public override token;
+
+    uint256 public redeemed;
+
+    Status public override status;
+
+    /**
+     * @dev Emitted when a DebtToken is redeemed for underlying tokens
+     * @param receiver Receiver of tokens
+     * @param burnedAmount Amount of DebtTokens burned
+     * @param redeemedAmount Amount of token received
+     */
+    event Redeemed(address receiver, uint256 burnedAmount, uint256 redeemedAmount);
+
+    /**
+     * @dev Emitted when loan gets liquidated
+     * @param status Final loan status
+     */
+    event Liquidated(Status status);
+
+    function initialize(
+        ITrueFiPool2 _pool,
+        address _lender,
+        address _borrower,
+        address _liquidator,
+        uint256 _debt
+    ) external initializer {
+        ERC20.__ERC20_initialize("TrueFi Debt Token", "DEBT");
+
+        pool = _pool;
+        token = _pool.token();
```

Comment:
Would it make sense to make `token()` a view that delegates to `pool.token()`?

I'm not sure how much we care about the gas saved from caching the `token` address in the loan itself. I don't have a strong opinion on this, just noticed a possibility to reduce contract state now that we're reimplementing. Please push back if you think it still makes sense to cache.